activerecord/lib/active_record/railtie.rb

```diff
@@ -178,6 +182,11 @@ class Railtie < Rails::Railtie # :nodoc:
     initializer "active_record.check_represent_sqlite3_boolean_as_integer" do
       config.after_initialize do
         ActiveSupport.on_load(:active_record_sqlite3adapter) do
+          sqlite3_represent_boolean_as_integer = Rails.application.config.active_record.delete(:sqlite3_represent_boolean_as_integer)
+          unless sqlite3_represent_boolean_as_integer.nil?
```

Comment:
Don't we have several of these kinds of ifs? Maybe it's time for a root cause analysis to either remove the need for them or a higher level abstraction?